frontend/app/src/components/section-components/chat/members-list.tsx

```diff
@@ -1,17 +1,55 @@
-import { User, UserListType } from '../../global-components/interface';
+import { channelRole, User, UserListType } from '../../global-components/interface';
 import UsersList from '../users-list';
+import { useEffect } from "react";
+import { useQueryClient, UseQueryResult } from "react-query";
+import { toast, ToastContainer } from "react-toastify";
+import { socket } from 'src/components/global-components/client-socket';
+import { useChannelRoles } from 'src/components/query-hooks/useGetChannels';
 
 interface MembersListProps {
   channelUsers: User[];
   user: User;
+  channelId: string;
 }
 
-function MembersList({ channelUsers, user }: MembersListProps) {
+function MembersList({ channelUsers, user, channelId }: MembersListProps) {
+  const customToastId = "custom-toast-error-role";
+  const roleQueryKey = 'rolesInChannel';
+  const queryClient = useQueryClient();
+
+  const roleUsers: UseQueryResult<{
+    userId: string;
```

Comment:
Might be cleaner to do the` useMyChannelRole(channelId)` here and invalidate the queryCache on reveiving a role updated. For other components requiring the data, we could use queryClient.